stl/inc/optional

```diff
@@ -11,15 +11,9 @@
 #if !_HAS_CXX17
 _EMIT_STL_WARNING(STL4038, "The contents of <optional> are available only with C++17 or later.");
 #else // ^^^ !_HAS_CXX17 / _HAS_CXX17 vvv
-#if _HAS_CXX20
-#include <compare>
-#endif // _HAS_CXX20
 #include <exception>
-#include <initializer_list>

```

Comment:
Is there a reason to remove this `#include`? It's still used down below and this is just making an explicit dependency implicit.

Reply:
Agreed. Our product code inclusions are relatively disorganized, but when we're using `Meow`, and it's supposed to be provided by `<meow_hdr>`, and there's no cheaper way (given our implementation knowledge) of getting `Meow`, explicitly including `<meow_hdr>` is the clearest thing to do. This remains true even if (given our implementation knowledge) we're guaranteed to drag in `<meow_hdr>` via `<woof>`.

Our code is far from uniformly following this principle (lots of code includes `<woof>` only and expects to get `Meow`), but avoiding backward movement would be nice.

Note that this specifically does not apply when we can use our implementation knowledge to avoid dragging in large parts of a public header - the classic example being `<xstring>` avoiding all of `<string>`.

Reply:
Agreed. I'm also adding conditional inclusion of `<compare>` to `<variant>` now for consistency.